Dockerfile

```
@@ -1,10 +1,11 @@
 FROM ruby:3.2.2-alpine3.18@sha256:198e97ccb12cd0297c274d10e504138f412f90bed50c36ebde0a466ab89cf526 AS build
+ARG NODEJS_VERSION=18
```

@AP-Hunt I couldn't decide whether to pull out the other version numbers as well; it occurred to me we could do something like:

```suggestion
ARG ALPINE_VERSION=3.18
ARG NODEJS_VERSION=18
ARG RUBY_VERSION=3.2.2

ARG DOCKER_IMAGE_DIGEST=sha256:198e97ccb12cd0297c274d10e504138f412f90bed50c36ebde0a466ab89cf526

FROM ruby:${RUBY_VERSION}-alpine${ALPINE_VERSION}@${DOCKER_IMAGE_DIGEST} AS build
```

It would make the `update_app_versions` script easier to write, but the Dockerfile harder to read. What do you think?

---

This still feels quite readable to me as a non-expert - and I think these version numbers are the bits that change the most, so it's nice to be able to see them at a glance.

---

I agree with David. The expression on line 7 is more complicated, but splitting the versions into their own args makes for overall simpler maintenance.

---

Thanks both, I've redone this commit, re-review would be appreciated 🙏